src/main/java/net/neoforged/neoforge/event/server/RegisterSchemaEvent.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.server;
+
+import java.util.Map;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.dedicated.DedicatedServer;
+import net.minecraft.server.jsonrpc.ManagementServer;
+import net.minecraft.server.jsonrpc.api.SchemaComponent;
+import net.neoforged.bus.api.Event;
+import org.jetbrains.annotations.ApiStatus.Internal;
+
+/**
+ * Event to register {@link SchemaComponent SchemaComponents} used in incoming and outgoing JSON-RPC methods.<br>
+ * Fired before the {@link ManagementServer} is instantiated in {@link DedicatedServer#initServer()}.
+ */
+public class RegisterSchemaEvent extends Event {
```

Comment:
`Schema` sounds a bit vague, maybe `RPCSchema`? Not sure, could use some input from others